applications/GeoMechanicsApplication/custom_utilities/process_utilities.cpp

```diff
@@ -16,6 +16,34 @@
 #include "containers/model.h"
 #include "includes/kratos_parameters.h"
 
+namespace
+{
+struct StringHash {
+    using is_transparent = void; // Enables heterogeneous operations.
+
+    std::size_t operator()(std::string_view sv) const
+    {
+        std::hash<std::string_view> hasher;
+        return hasher(sv);
+    }
+};
+
+void ExtractModelPartNames(const auto& rProcessList,
+                           std::unordered_set<std::string, StringHash, std::equal_to<>>& rDomainConditionNames,
+                           std::string_view RootName,
+                           std::string_view Prefix)
+{
+    for (const auto& r_process : rProcessList) {
+        if (r_process.Has("Parameters") && r_process["Parameters"].Has("model_part_name")) {
+            auto model_part_name = r_process["Parameters"]["model_part_name"].GetString();
+            if (model_part_name == RootName) continue;
+            if (model_part_name.rfind(Prefix, 0) == 0) model_part_name.erase(0, Prefix.size());
```

Comment:
If you want to check whether `model_part_name` starts with `Prefix` (and then remove the `Prefix` if it does), I would use member [`starts_with`](https://en.cppreference.com/w/cpp/string/basic_string/starts_with.html) (available since C++20):
```suggestion
            if (model_part_name.starts_with(Prefix) model_part_name.erase(0, Prefix.size());
```
I feel this is much easier to read.

Reply:
agreed, done